bpi.syndicate.inc

```diff
@@ -332,149 +454,181 @@ function bpi_preprocess_bpi_search_results(&$variables) {
- *   An array, in structure:
- *   - key: value
+ * @return string
+ *   The link for sorting.
  */
-function _bpi_build_query($key = '', $value = '', $ignore = NULL) {
-  $destination = drupal_get_destination();
-  $current_path = parse_url(reset($destination));
-  $query = isset($current_path['query']) ? $current_path['query'] : '';
-
-  $new_query_pairs = array();
-
-  if (!empty($query)) {
-    // Get the parameters array.
-    $components = explode('&', $query);
-
-    // Iterate and rearrange them.
-    foreach ($components as $component) {
-      $temp = explode('=', $component);
-
-      // Ignore a key from query string.
-      if ($temp[0] == $ignore) {
-        continue;
-      }
-
-      // It's required to decode some values here, since Drupal's url()
-      // encodes it weird, munging the 'space' entity value (%20).
-      $new_query_pairs[$temp[0]] = urldecode($temp[1]);
-    }
+function bpi_get_sort_link($text, $field, $default_direction = 'asc') {
+  $current_sorting = _bpi_get_current_sorting($field);
+  if ($current_sorting) {
+    $sort_direction = $current_sorting == 'asc' ? 'desc' : 'asc';
+  }
+  else {
+    $sort_direction = $default_direction;
   }
+  $query = _bpi_build_query(array(
+    BPI_QUERY_SORT_NAME => array(
+      $field => $sort_direction,
+    ),
+  ), array(BPI_QUERY_SORT_NAME));
+  $image = $current_sorting ? theme('tablesort_indicator', array('style' => $sort_direction)) : '';
+  return l($text . $image, current_path(), array('query' => $query, 'html' => TRUE));
+}
 
-  // Append the new key/value pair to existing ones.
-  if (!empty($key) && !empty($value) && $key != $ignore) {
-    $new_query_pairs[$key] = $value;
+/**
+ * Get current sorting or direction.
+ *
+ * @param string $field
+ *   Field to get sorting direction for.
+ *
+ * @return string|array
+ *   The current sorting (field => direction) or the direction for a specified
+ *   field.
+ */
+function _bpi_get_current_sorting($field = NULL) {
+  $sorting = _bpi_get_query_parameter(BPI_QUERY_SORT_NAME);
+  if ($field) {
+    return isset($sorting[$field]) ? $sorting[$field] : NULL;
   }
+  return $sorting;
+}
 
-  return $new_query_pairs;
+/**
+ * Get current search.
+ *
+ * @return array
+ *   The search.
+ */
+function _bpi_get_current_search_query() {
+  $text = _bpi_get_query_parameter('text', NULL);
+  $amount = _bpi_get_query_parameter('amount');
+  $query = array();
+  if ($text) {
+    $query['text'] = $text;
+  }
+  if ($amount) {
+    $query['amount'] = $amount;
+  }
+  return $query;
 }
 
 /**
- * Used to retrieve the searched phrase after submit is clicked.
+ * Get current filter.
  *
- * @see bpi_syndicate_page_form()
+ * @return array
+ *   The filter.
+ */
+function _bpi_get_current_filter() {
+  $value = _bpi_get_query_parameter('filter', array());
+  // @TODO: Make sure that only valid filter parameters are present.
+  if (isset($value['agency_internal'])) {
+    $value['agencyInternal'] = $value['agency_internal'];
```

Comment:
A comment about this change between "agencyInternal" and "agency_internal", would be nice. As it may be seen of others as an mistake.